test/gmp.jl

```diff
@@ -441,8 +441,38 @@ end
 @test isqrt(big(4)) == 2
 @test isqrt(big(5)) == 2
 
-@test big(5)^true == big(5)
-@test big(5)^false == one(BigInt)
+
+@testset "Exponentiation operator" begin
+    @test big(5)^true == big(5)
+    @test big(5)^false == one(BigInt)
+    # Test BigInt and Int8 exponentiation are consistent.
+    for i = typemin(Int8):typemax(Int8), j = typemin(Int8):typemax(Int8)
```

Comment:
Do we really want to test 2^16 different `BigInt` calculations? Also, this would be good to have in a `@testset for` instead.